src/Azure/Azure.Quantum.Client/Storage/StorageHelper.cs

```diff
@@ -81,14 +120,27 @@ public StorageHelper(string connectionString)
             TimeSpan expiryInterval,
             SharedAccessBlobPermissions permissions)
         {
-            SharedAccessBlobPolicy adHocSAS = CreateSharedAccessBlobPolicy(expiryInterval, permissions);
+            try
+            {
+                SharedAccessBlobPolicy adHocSAS = CreateSharedAccessBlobPolicy(expiryInterval, permissions);
 
-            CloudBlob blob = this.storageAccount
-                .CreateCloudBlobClient()
-                .GetContainerReference(containerName)
-                .GetBlobReference(blobName);
+                CloudBlob blob = this.storageAccount
+                    .CreateCloudBlobClient()
+                    .GetContainerReference(containerName)
+                    .GetBlobReference(blobName);
+
+                return blob.Uri + blob.GetSharedAccessSignature(adHocSAS);
+            }
+            catch (Exception ex)
+            {
+                throw new StorageClientException(
+                    "Could not get BLOB sas URI",
```

Comment:
```suggestion
                    "Could not get BLOB shared access signature URI",
```

(Alternatively, "SAS" should be capitalized, but that is a lot of acronyms in a row.)